packages/shared/src/contexts/PaymentContext.tsx

```diff
@@ -155,18 +161,34 @@ export const PaymentContextProvider = ({
     enabled: !!paddle && !!planTypes && !!geo,
   });
 
-  const productOptions = useMemo(
+  const productOptions: Array<ProductOption> = useMemo(
     () =>
-      productPrices?.data?.details?.lineItems?.map((item) => ({
-        label: item.price.description,
-        value: item.price.id,
-        price: item.formattedTotals.total,
-        priceUnformatted: Number(item.totals.total),
-        currencyCode: productPrices?.data.currencyCode as string,
-        extraLabel: item.price.customData?.label as string,
-        appsId: item.price.customData?.appsId as string,
-      })) ?? [],
-    [productPrices?.data.currencyCode, productPrices?.data?.details?.lineItems],
+      productPrices?.data?.details?.lineItems?.map((item) => {
```

Comment:
Can you explain what was the change here? Crucial part of the resolving but it is a bit hard to see what is going on.

Reply:
We are showing the monthly price now, so I've added the properties and grouped into `option.price.*`. Also need the duration data in order to get the annual in onboarding.

Reply:
Also Paddle do not provide the formatted monthly price for annual subscriptions so I've to build it

Reply:
Got it, left to comments on the logic additionally.